config/config.go

```diff
@@ -171,10 +170,42 @@ func WriteConfigValueToConfigFile(key, value string) error {
 	}
 	configValues[key] = value
 
-	err = WriteYamlFile(f.Name(), &configValues)
+	err = atomicwrite(f.Name(), &configValues)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func atomicwrite(file string, c *ContextMap) (err error) {
```

Comment:
So, the issue this is fixing, is that when using multiple processes (by invoking the Fn CLI from a shell as multplie times as background processes) this config file can attempt to be read by one process as another is writing to it. The config file  contains the information on what 'context' the CLI should be using, so in these situations comes back 'empty' which is the definition of the `default` context.

The solution presented here is to write to a tmp file first, then, move it. Subsequently make the file write an atomic operation.

Nice. Looks goo to me.